SDWebImage/SDWebImageDownloader.m

```diff
@@ -135,7 +135,10 @@ - (void)setOperationClass:(Class)operationClass {
                                                          progress:^(NSInteger receivedSize, NSInteger expectedSize) {
                                                              SDWebImageDownloader *sself = wself;
                                                              if (!sself) return;
-                                                             NSArray *callbacksForURL = [sself callbacksForURL:url];
+                                                             __block NSArray *callbacksForURL;
+                                                             dispatch_sync(sself.barrierQueue, ^{
+                                                                 callbacksForURL = sself.URLCallbacks[url];
+                                                             });
```

Comment:
@timkoma Yes, you are right, and it will cause a crash in fast enumeration in case items are removed from `callbacksForURL`.
I will fix it. Thanks :+1: